stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -294,3 +294,36 @@ function eigvals!(A::StridedMatrix{T}, F::LU{T,<:StridedMatrix}; sortby::Union{F
     rdiv!(A, U)
     return eigvals!(A; sortby)
 end
+
+
+function eigen(A::Hermitian{Complex{T}, <:Tridiagonal}; kwargs...) where {T}
+    (; dl, d, du) = parent(A)
+    N = length(d)
+    if N <= 1
+        eigen(parent(A); kwargs...)
+    else
+        if A.uplo == 'U'
+            E = du'
```

Comment:
It should not be necessary to allocate new vector `E` (btw. `du'` is a `Matrix` with one row).
The necessary operations to populate `S` should be possible  on the fly.

Reply:
Hi Klaus, no new allocations are made for `E`, since `du'` is lazy. The vector `Er` needs to be allocated though since it's used to construct `B = SymTridiagonal(real.(d), Er)` which is then passed to `eigen`.